test/test_dh.c

```diff
@@ -340,7 +340,65 @@ int test_dh_pgen_pkey(ENGINE *e, void *data)
     return err;
 }
 
-int test_dh_pkey(ENGINE *e, void *data)
+#if !defined(WE_SINGLE_THREADED) && defined(_WIN32)
+
+typedef struct {
+    ENGINE* e;
+    EVP_PKEY* params;
+} DH_KEYGEN_THREAD_VARS;
+
+static DWORD WINAPI DhKeyGenThreadFunc(LPVOID arg)
+{
+    DH_KEYGEN_THREAD_VARS* vars = (DH_KEYGEN_THREAD_VARS*)arg;
+
+    return test_dh_pgen_pkey(vars->e, vars->params);
+}
+
+int test_dh_key_gen_multithreaded(ENGINE* e, EVP_PKEY* params)
```

Comment:
Can you add a comment above this explaining that this is a regression test for \<describe problem\>?

Reply:
Certainly.  Added in new commit.